plugins/GSdx/Renderers/Common/GSTextureCache.cpp

```diff
@@ -1451,13 +1454,13 @@ GSTextureCache::Source* GSTextureCache::CreateSource(const GIFRegTEX0& TEX0, con
 		if (m_paltex && psm.pal > 0)
 		{
 			src->m_texture = m_renderer->m_dev->CreateTexture(tw, th, Get8bitFormat());
-			src->m_should_have_tex_palette = true;
-			// Palette (Palette object, clut copy and palette texture) is going to be attached in Source lookup code exploiting initial condition !src->m_clut
+			AttachPaletteToSource(src, psm.pal, true);
 		}
 		else {
 			src->m_texture = m_renderer->m_dev->CreateTexture(tw, th);
-			// by default: src->m_should_have_tex_palette = false;
-			AttachPaletteToSource(src, psm.pal, false); // Attach only Palette object and clut copy
+			if (psm.pal > 0) {
```

Comment:
This change is most likely what fixed #2702, as we were malloc'in 0 bytes clut arrays without the `psm.pal > 0` check.

Reply:
No, that change doesn't seem to affect `Tales of Legendia`.

Reply:
Oh strange. Could you please try again the last version of PR plugin to see if the regression is still fixed? I applied some modifications to make things easier to manage.

Reply:
Yes, I applied the whole PR as a patch to master and both `Tales of Legendia` and `Tales of the Abyss` still seem fixed with and without `8 bits textures`.